### DISCUSSION TITLE
Should the `Televerse.filter` method be generic method?

### DISCUSSION THREAD

**Discussion by The0xArchitect** [HeySreelal/televerse#71] [Polls]
Hey! Just have a question about a weird idea. 

If you have already tried you might already know the `Televerse.filter` method is highly effective for filtering out and listening to highly specific message updates. But as I've already said, this method only listens to `Message` updates.

That is the context passed to the callback and the predicate is actually of type MessageContext, that is, you cannot listen to an inline query update or a callback query update with the filter method.

What I'm thinking is an approach, something like the following code, by which you can match any update by passing the generic type to the filter method.

```dart

void main()  {
  Bot bot = Bot("TOKEN");

  bot.filter<InlineQueryContext>((ctx) {
    return ctx.query == "hello";
  }, (ctx) {
    ctx.answer([
      InlineQueryResultArticle(
        id: "1",
        title: "Hello",
        inputMessageContent: InputTextMessageContent("Hello World!"),
      )
    ]);
  });


  bot.filter<MessageContext>((ctx) {
    return ctx.message.text == "hello";
  }, (ctx) {
    ctx.reply("Hello World!");
  });

  bot.filter<CallbackQueryContext>((ctx) {
    return ctx.data == "hello";
  }, (ctx) {
    ctx.answer("Hello World!");
  });
}
```

**Comment by im-trisha** [comment]
Yes this could probably be a nice thing to add
I finally got some of the projects I had to do done, I can finally help you, you know how to get in touch with me already, if you need any help ❤️

**Reply by The0xArchitect** [reply]
Whoohoo! I'm more excited to see you being active on the project! 🎉
